apps/challenges/views.py

```diff
@@ -340,3 +361,301 @@ def challenge_phase_split_list(request, challenge_pk):
     serializer = ChallengePhaseSplitSerializer(result_page, many=True)
     response_data = serializer.data
     return paginator.get_paginated_response(response_data)
+
+
+from .utils import get_file_content
+@throttle_classes([UserRateThrottle])
+@api_view(['POST'])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def create_challenge_using_zip_file(request, challenge_host_team_pk):
+    """
+    Creates a challenge using a zip file.
+    """
+    challenge_host_team = get_challenge_host_team_model(challenge_host_team_pk)
+
+    serializer = ChallengeConfigSerializer(data=request.data, context={'request': request})
+    if serializer.is_valid():
+        uploaded_zip_file = serializer.save()
+        uploaded_zip_file_path = serializer.data['zip_configuration']
+    else:
+        response_data = serializer.errors
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    # All files download and extract location.
+    BASE_LOCATION = tempfile.mkdtemp()
+    try:
+        response = requests.get(uploaded_zip_file_path, stream=True)
+        unique_folder_name = ''.join([random.choice(string.ascii_letters + string.digits) for i in xrange(10)])
+        CHALLENGE_ZIP_DOWNLOAD_LOCATION = join(BASE_LOCATION, '{}.zip'.format(unique_folder_name))
+        try:
+            if response and response.status_code == 200:
+                with open(CHALLENGE_ZIP_DOWNLOAD_LOCATION, 'w') as zip_file:
+                    zip_file.write(response.content)
+        except IOError:
+            response_data = {
+                'error': 'Unable to process the uploaded zip file. Please upload it again! '
+                }
+            return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    except requests.exceptions.RequestException:
+        response_data = {
+            'error': 'A server error occured while processing zip file. Please try uploading it again.'
```

Comment:
Replace with `'error': 'A server error occurred while processing zip file. Please try uploading it again!'`